### PR TITLE
Move doxygen generation to nightly only

### PR DESCRIPTION
**Pull request by janeyx99**
as it is OOMing yet again (8 out of 28 commits on friday)

a milder version of https://github.com/pytorch/pytorch/pull/80451

### CODE REVIEW

**Review by jbschlosser — APPROVED** [review]
LGTM! But I'm not too familiar with CI-side stuff :)

**Review by ZainRizvi — APPROVED** [review]

File: docs/cpp/source/conf.py [hunk 1]
```diff
@@ -26,15 +29,17 @@
...
+run_doxygen = os.environ.get('RUN_DOXYGEN', "false") == "true"
 
 # Add any Sphinx extension module names here, as strings. They can be
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
 # ones.
 extensions = [
     'sphinx.ext.intersphinx',
+] + ([
     'breathe',
     'exhale'
```
Comment (ZainRizvi) [inline docs/cpp/source/conf.py:41]: Is merely adding these two extensions is all that's needed to run doxygen?

Comment (janeyx99) [reply docs/cpp/source/conf.py:41]: https://github.com/pytorch/pytorch/runs/7615170554?check_suite_focus=true seems so!

**Review by malfet — APPROVED** [review]

**Review by malfet — COMMENTED** [review]
I've considered requesting changes, but then realized it's probably fine and can be done in follow-up PR/issues:
- This PR regresses logic for release builds, where cpp docs will be rebuilt without doxygen
- It creates a config that developer can not replicate on PR without modifying .yml files

**Pull request closed without merging**